packages/block-editor/src/components/block-list/style.scss

```diff
@@ -104,8 +100,28 @@
 			border-color: var(--wp-admin-theme-color);
 		}
 	}
-}
 
+	// Ensure an accurate partial text selection.
+	// To do this, we disable text selection on the main container, then re-enable it only on the
+	// elements that actually get selected.
+	// To keep in mind: user-select is currently inherited to all nodes inside.
+	user-select: none;
```

Comment:
Have you made any experiments with `auto` value([ref](https://developer.mozilla.org/en-US/docs/Web/CSS/user-select#values))? I haven't thought much about it, but this effects everything besides the exceptions below. For example you can't select any placeholder text etc.. - maybe that's better though..

Reply:
Thank you for looking 🙏

`auto` is default, and as it turns out, is the cause of the glitchy selection we've been seeing in Safari. For a partial selection between two paragraphs, the following selection is accurate to what you are actually selecting:

<img width="843" alt="Screenshot 2022-08-22 at 13 09 11" src="https://user-images.githubusercontent.com/1204802/185907364-c1521eb5-c080-46d5-98bb-125764ea9b8e.png">

The above works in this branch because we disallow everything but the paragraph text to be selected. It seems Safari's interpretation of selection in browsers is to capture everything, every div, every pseudo element. And so if we don't disable user select, we get the trunk behavior, which is this:

<img width="1867" alt="Screenshot 2022-08-22 at 13 06 45" src="https://user-images.githubusercontent.com/1204802/185907528-0b1d901b-ba53-44ec-8f1f-ddca16da5419.png">

So in order to accurately depict to the user what's selected (and what they can copy if they press ⌘C), we have to disable user-select on everything else. The fact that this affects everything down the stack is the main thing to test and write exceptions for, and I'm not sure if there's a better way.